ament_cmake_core/cmake/index/ament_index_register_package.cmake

```diff
@@ -29,5 +29,5 @@ function(ament_index_register_package)
   endif()
 
   # register package name with the resource type 'packages'
-  ament_index_register_resource("packages" ${ARGN})
+  ament_index_register_resource("packages" ${ARG_PACKAGE_NAME})
```

Comment:
The check above is wrong. The CMake function currently doesn't allow to pass a package name. Instead the check should test for "unused arguments".

The call to `ament_index_register_resource` can either just pass `${ARGN}` as before or explicitly `"PACKAGE_NAME" "${ARG_PACKAGE_NAME}"`.